frontend/src/hooks/useActions.ts

```diff
@@ -588,4 +522,23 @@ const promptLoadFile = (parse, onData, errorMessage = 'Failed to parse file') =>
   input.click()
 }
 
+// Takes in the IDs of states, comments, and transitions
+// Parameters also include  the current project and whether a template is being created
+// Outputs a CopyData to be copied or Template object to be created into a template
+export const selectionToCopyTemplate = (stateIds: number[], commentIds: number[], transitionIds: number[], project): CopyData => {
+  const selectedStates: AutomataState[] = project.states.filter(state => stateIds.includes(state.id))
+  const selectedComments: ProjectComment[] = project.comments.filter(comment => commentIds.includes(comment.id))
+  const selectedTransitions = project.transitions.filter(transition => transitionIds.includes(transition.id))
+  const isInitialSelected = stateIds.includes(project.initialState)
+  const returnObject: CopyData = {
+    states: selectedStates,
+    comments: selectedComments,
+    transitions: selectedTransitions,
+    projectSource: project._id,
+    projectType: project.projectType,
+    initialStateId: isInitialSelected ? project.initialState : null
+  }
+  return returnObject
```

Comment:
```suggestion
  return {
    states: selectedStates,
    comments: selectedComments,
    transitions: selectedTransitions,
    projectSource: project._id,
    projectType: project.projectType,
    initialStateId: isInitialSelected ? project.initialState : null
  }
```
Very small nitpick 😄 I just usually like to return objects directly

Reply:
Nah that's all good it makes sense, I think this was left over from when there were some different options of stuff to assign to the return